src/evo/providertx.cpp

```diff
@@ -246,3 +246,20 @@ void ProRegPL::ToJson(UniValue& obj) const
     obj.pushKV("operatorReward", (double)nOperatorReward / 100);
     obj.pushKV("inputsHash", inputsHash.ToString());
 }
+
+bool GetProRegCollateral(const CTransactionRef& tx, COutPoint& outRet)
+{
+    if (tx == nullptr) {
```

Comment:
this pointer cannot be null.

Reply:
Yeah. For now this is only reached via `AddToWalletIfInvolvingMe` --> `LockIfMyCollateral`, so we know that `ptx` is not null. But, imo, it doesn't hurt to have this check at the beginning of the function, in case it gets used elsewhere in the future.